storage/range.go

```diff
@@ -387,11 +387,16 @@ func (r *Range) redirectOnOrAcquireLeaderLease(timestamp proto.Timestamp) error
 	return nil
 }
 
-// verifyLeaderLease checks whether the requesting replica (by raft
-// node ID) holds the leader lease covering the specified timestamp.
-func (r *Range) verifyLeaderLease(originRaftNodeID proto.RaftNodeID, timestamp proto.Timestamp) bool {
-	l := r.getLease()
-	return originRaftNodeID == l.RaftNodeID && timestamp.Less(l.Expiration)
+// verifyLeaderLease checks whether the requesting replica (by raft node ID)
+// holds the leader lease covering the specified timestamp.
+func verifyLeaderLease(lease *proto.Lease, replicaRaftNodeID proto.RaftNodeID, timestamp proto.Timestamp) (bool, bool) {
+	if lease == nil || lease.RaftNodeID == 0 {
+		// The lease has never been held.
+		return false, true
+	}
+	held := lease.RaftNodeID == replicaRaftNodeID
+	expired := !timestamp.Less(lease.Expiration)
```

Comment:
no, but it doesn't really matter and the refactor you suggested above (which I'm in the middle of) suggests that it's better (more comfortable) to exclude the endpoint, so I'll likely make that change.